rasa_core/channels/slack.py

```diff
@@ -54,6 +57,7 @@ def send_attachment(self, recipient_id, attachment, message=""):
     def _convert_to_slack_buttons(buttons):
         return [{"text": b['title'],
                  "name": b['payload'],
+                 "value": b['payload'],
```

Comment:
why `name` and `value` are both set to `b['payload']`?

Reply:
because slack has three values for its buttons, and the templates for rasa buttons assume only two. feel free to mess with this as you see fit. 
Incidently I have changed the send text with buttons function so it does not get the null call back id if the user has not set any message text (instead it creates a presumably unique call back id using the buttons):

    def send_text_with_buttons(self, recipient_id, message, buttons, **kwargs):
	recipient = self.slack_channel or recipient_id

        if len(buttons) > 5:
            logger.warning("Slack API currently allows only up to 5 buttons. "
                           "If you add more, all will be ignored.")
            return self.send_text_message(recipient, message)

        if(len(message) > 0):
            callback_string = message.replace(' ', '_')[:20];
        else:
            callback_string = self._get_text_from_slack_buttons(buttons);
            callback_string = callback_string.replace(' ', '_')[:20];


        button_attachment = [{"fallback": message,
                              "callback_id": callback_string,
                              "actions": self._convert_to_slack_buttons(
                                  buttons)}]

        super(SlackBot, self).api_call("chat.postMessage",
                                       channel=recipient,
                                       as_user=True,
                                       text=message,
                                       attachments=button_attachment)